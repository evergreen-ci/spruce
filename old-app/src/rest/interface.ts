import { AxiosPromise } from "axios";
import * as evergreen from "evergreen.js";
import * as models from "evergreen.js/lib/models";
import * as mock from "./evergreen_mock";
import { isDevelopment, getApiUrl, getUiUrl, isTest } from "../utils";

export interface Evergreen {
  apiURL: string;
  uiURL: string;
  username?: string;
  key?: string;

  getDistros: () => AxiosPromise<any>;
  getRecentTasks: (
    verbose?: boolean,
    lookbackMins?: number,
    status?: string
  ) => AxiosPromise<any>;
  getToken: (username?: string, password?: string) => AxiosPromise<any>;
  getPatches: (
    username?: string,
    page?: number
  ) => AxiosPromise<models.Patches>;
  getProjectPatches: (
    projectName?: string,
    page?: number
  ) => AxiosPromise<models.Patches>;
  getLogs: (
    taskId: string,
    type: string,
    executionNumber: number
  ) => AxiosPromise<string>;
  getBuild: (id: string) => AxiosPromise<models.Build>;
  getTasksForBuild: (buildId: string) => AxiosPromise<models.APITask[]>;
  getTestsForTask: (taskId: string) => AxiosPromise<models.APITest[]>;
  getAdminConfig: () => AxiosPromise<models.AdminSettings>;
  setAdminConfig: (
    settings: models.AdminSettings
  ) => AxiosPromise<models.AdminSettings>;
  getBanner: () => AxiosPromise<any>;
}

interface EvergreenClientProps {
  username?: string;
  key?: string;
}

export function EvergreenClient({
  username = "",
  key = ""
}: EvergreenClientProps = {}): Evergreen {
  const apiURL = getApiUrl();
  const uiURL = getUiUrl();

  if (isDevelopment() || isTest()) {
    return new mock.client(apiURL, uiURL, username, key);
  }
  return new evergreen.client(apiURL, uiURL, username, key);
}
