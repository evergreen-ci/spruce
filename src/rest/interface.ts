import * as evergreen from "evergreen.js";
import * as request from "request";
import * as mock from "./evergreen_mock"

export interface Evergreen {
  apiURL: string;
  uiURL: string;
  username?: string;
  key?: string;

  getDistros: (callback: request.RequestCallback) => void;
  getRecentTasks: (callback: request.RequestCallback, verbose?: boolean, lookbackMins?: number, status?: string) => void;
  getToken: (callback: request.RequestCallback, username?: string, password?: string) => void;
  getPatches: (callback: request.RequestCallback, username?: string, page?: number) => void;
  getLogs: (callback: request.RequestCallback, taskId: string, type: string, executionNumber: number) => void;
  getBuild: (callback: request.RequestCallback, id: string) => void;
  getTasksForBuild: (callback: request.RequestCallback, buildId: string) => void;
  getTestsForTask: (callback: request.RequestCallback, taskId: string) => void;
  getAdminConfig: (callback: request.RequestCallback) => void;
  setAdminConfig: (callback: request.RequestCallback, settings: any) => void;
  getBanner: (callback: request.RequestCallback) => void;
}

export function EvergreenClient(apiURL: string, uiURL: string, username?: string, key?: string, isMock: boolean = false): Evergreen {
  if (isMock) {
    return new mock.client(apiURL, uiURL, username, key);
  }
  return new evergreen.client(apiURL, uiURL, username, key);
}