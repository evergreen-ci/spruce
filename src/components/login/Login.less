.login-button {
  padding: 10px;
}
