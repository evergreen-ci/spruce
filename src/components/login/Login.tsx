import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField, Typography } from '@material-ui/core';
import * as React from 'react';
import * as rest from "../../rest/interface";
import '../../styles.css';

export const UserContext = React.createContext({
  username: "",
});
export const UserContextProvider = UserContext.Provider;
export const UserContextConsumer = UserContext.Consumer;

interface State {
  open: boolean;
  username: string
  password: string
  errorText: string
}

class Props {
  public client: rest.Evergreen;
  public updateUsername: (username: string) => void;
}

export class Login extends React.Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.state = {
      open: false,
      username: "",
      password: "",
      errorText: ""
    };
  }

  public render() {

    return (
      <div>
        <Button onClick={this.handleClickOpen} color="inherit" className="login-button" id="login-button">
          {this.context.username === "" ? "Log In" : "Log Out"}
        </Button>
        <Dialog open={this.state.open} onClose={this.handleClickClose} id="login-modal">
          <DialogTitle id="form-dialog-title">Log In</DialogTitle>
          <DialogContent>
            <DialogContentText>
              Please enter your username and password.
            </DialogContentText>
            <TextField
              autoFocus={false}
              margin="dense"
              id="username"
              label="username"
              type="string"
              fullWidth={true}
              required={true}
              onChange={this.updateSingleField(this.state, "username")}
              value={this.state.username}
            />
            <TextField
              autoFocus={false}
              margin="dense"
              id="password"
              label="password"
              type="password"
              fullWidth={true}
              required={true}
              onChange={this.updateSingleField(this.state, "password")}
              value={this.state.password}
            />
            <Typography color="error">
              {this.state.errorText}
            </Typography>
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClickClose} color="primary" id="cancel-button">
              Cancel
            </Button>
            <Button onClick={this.handleClickSubmit} color="primary" id="submit-button">
              Submit
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }

  private handleClickOpen = () => {
    if (this.context.username === "") {
      this.setState({
        open: true,
      });
    } else {
      this.setState({
        username: "",
        password: ""
      });
      this.props.updateUsername("");
      this.context.username = "";
    }
  }

  private handleClickClose = () => {
    this.setState({
      open: false,
      username: "",
      password: "",
      errorText: ""
    });
  }

  private handleClickSubmit = () => {
    this.props.client.getToken((err, resp, body) => {
      if (err || resp.statusCode >= 300) {
        if (resp.statusCode === 401) {
          this.setState({
            errorText: "Invalid username or password"
          });
        } else {
          console.log("got error " + err + " with status code " + resp.statusCode);
        }
      } else {
        this.props.updateUsername(this.state.username);
        this.context.username = this.state.username;
        this.setState({
          open: false,
        });
      }
    }, this.state.username, this.state.password);
  }

  private updateSingleField = <T, K extends keyof T>(state: T, key: K) => {
    return (value: React.ChangeEvent<HTMLInputElement>) => {
      const newState = Object.assign({}, state);
      newState[key] = value.target.value as any;
      this.setState(newState);
    }
  }
}

Login.contextType = UserContext;

export default Login;