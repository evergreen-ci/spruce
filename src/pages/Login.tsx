import React from "react";

export const Login: React.FC = () => {
  return <div>Login</div>;
};
