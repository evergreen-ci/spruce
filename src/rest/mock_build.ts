export function getMockBuild() {
  return {
    "_id": "spruce_ubuntu1604_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
    "project_id": "spruce",
    "create_time": "2019-07-12T18:21:22.913Z",
    "start_time": "2019-07-12T18:30:09.414Z",
    "finish_time": "2019-07-12T18:32:17.147Z",
    "version": "5d28cfa05623434037b0294c",
    "branch": "spruce",
    "git_hash": "e44b6da8831497cdd4621daf4c62985f0c1c9ca9",
    "build_variant": "ubuntu1604",
    "status": "success",
    "activated": false,
    "activated_by": "github_pull_request",
    "activated_time": "2019-07-12T18:33:08.026Z",
    "order": 94,
    "task_cache": [
      {
        "id": "spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "display_name": "compile",
        "status": "success",
        "task_end_details": {
          "status": "success",
          "type": "test",
          "desc": "'subprocess.exec' in \"npm-build\" (#1)"
        },
        "start_time": "2019-07-12T18:30:09.414Z",
        "time_taken": 127414241328,
        "activated": true
      },
      {
        "id": "spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "display_name": "lint",
        "status": "success",
        "task_end_details": {
          "status": "success",
          "type": "test",
          "desc": "'subprocess.exec' in \"npm-lint\" (#1)"
        },
        "start_time": "2019-07-12T18:30:11.561Z",
        "time_taken": 53705398637,
        "activated": true
      },
      {
        "id": "spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "display_name": "test",
        "status": "success",
        "task_end_details": {
          "status": "success",
          "type": "test",
          "desc": "'attach.xunit_results' in \"attach-results\" (#1)"
        },
        "start_time": "2019-07-12T18:30:38.038Z",
        "time_taken": 49443215767,
        "activated": true
      }
    ],
    "tasks": [
      "spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
      "spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
      "spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22"
    ],
    "time_taken_ms": 127733,
    "display_name": "Ubuntu 16.04",
    "predicted_makespan_ms": 127414,
    "actual_makespan_ms": 127414,
    "origin": "patch",
    "status_counts": {
      "succeeded": 3,
      "failed": 0,
      "started": 0,
      "undispatched": 0,
      "inactive": 0,
      "dispatched": 0,
      "timed_out": 0
    }
  } as object
}