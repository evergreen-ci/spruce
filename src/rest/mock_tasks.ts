export function getMockTasks() {
  return (
    [
      {
        "task_id": "spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "project_id": "spruce",
        "create_time": "2019-07-08T18:39:15.000Z",
        "dispatch_time": "2019-07-12T18:30:08.667Z",
        "scheduled_time": "2019-07-12T18:29:49.412Z",
        "start_time": "2019-07-12T18:30:09.414Z",
        "finish_time": "2019-07-12T18:32:16.828Z",
        "ingest_time": "2019-07-12T18:21:23.024Z",
        "version_id": "5d28cfa05623434037b0294c",
        "revision": "e44b6da8831497cdd4621daf4c62985f0c1c9ca9",
        "priority": 0,
        "activated": true,
        "activated_by": "",
        "build_id": "spruce_ubuntu1604_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "distro_id": "ubuntu1604-test",
        "build_variant": "ubuntu1604",
        "depends_on": null,
        "display_name": "compile",
        "host_id": "sir-dah86y8k",
        "restarts": 0,
        "execution": 0,
        "order": 94,
        "status": "success",
        "status_details": {
          "status": "success",
          "type": "test",
          "desc": "'subprocess.exec' in \"npm-build\" (#1)",
          "timed_out": false
        },
        "logs": {
          "all_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=ALL",
          "task_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=T",
          "agent_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=E",
          "system_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_compile_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=S"
        },
        "time_taken_ms": 127414,
        "expected_duration_ms": 118661,
        "est_wait_to_start_ms": 0,
        "estimated_cost": 0.002576987475308642,
        "generate_task": false,
        "generated_by": "",
        "artifacts": null,
        "display_only": false,
        "mainline": false
      },
      {
        "task_id": "spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "project_id": "spruce",
        "create_time": "2019-07-08T18:39:15.000Z",
        "dispatch_time": "2019-07-12T18:30:10.725Z",
        "scheduled_time": "2019-07-12T18:29:49.412Z",
        "start_time": "2019-07-12T18:30:11.561Z",
        "finish_time": "2019-07-12T18:31:05.266Z",
        "ingest_time": "2019-07-12T18:21:23.063Z",
        "version_id": "5d28cfa05623434037b0294c",
        "revision": "e44b6da8831497cdd4621daf4c62985f0c1c9ca9",
        "priority": 0,
        "activated": true,
        "activated_by": "",
        "build_id": "spruce_ubuntu1604_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "distro_id": "ubuntu1604-test",
        "build_variant": "ubuntu1604",
        "depends_on": null,
        "display_name": "lint",
        "host_id": "sir-tzgr48xh",
        "restarts": 0,
        "execution": 0,
        "order": 94,
        "status": "success",
        "status_details": {
          "status": "success",
          "type": "test",
          "desc": "'subprocess.exec' in \"npm-lint\" (#1)",
          "timed_out": false
        },
        "logs": {
          "all_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=ALL",
          "task_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=T",
          "agent_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=E",
          "system_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_lint_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=S"
        },
        "time_taken_ms": 53705,
        "expected_duration_ms": 48963,
        "est_wait_to_start_ms": 0,
        "estimated_cost": 0.001086200200617284,
        "generate_task": false,
        "generated_by": "",
        "artifacts": null,
        "display_only": false,
        "mainline": false
      },
      {
        "task_id": "spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "project_id": "spruce",
        "create_time": "2019-07-08T18:39:15.000Z",
        "dispatch_time": "2019-07-12T18:30:37.337Z",
        "scheduled_time": "2019-07-12T18:29:49.412Z",
        "start_time": "2019-07-12T18:30:38.038Z",
        "finish_time": "2019-07-12T18:31:27.481Z",
        "ingest_time": "2019-07-12T18:21:23.095Z",
        "version_id": "5d28cfa05623434037b0294c",
        "revision": "e44b6da8831497cdd4621daf4c62985f0c1c9ca9",
        "priority": 0,
        "activated": true,
        "activated_by": "",
        "build_id": "spruce_ubuntu1604_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22",
        "distro_id": "ubuntu1604-test",
        "build_variant": "ubuntu1604",
        "depends_on": null,
        "display_name": "test",
        "host_id": "sir-prc87qqg",
        "restarts": 0,
        "execution": 0,
        "order": 94,
        "status": "success",
        "status_details": {
          "status": "success",
          "type": "test",
          "desc": "'attach.xunit_results' in \"attach-results\" (#1)",
          "timed_out": false
        },
        "logs": {
          "all_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=ALL",
          "task_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=T",
          "agent_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=E",
          "system_log": "https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_test_patch_e44b6da8831497cdd4621daf4c62985f0c1c9ca9_5d28cfa05623434037b0294c_19_07_12_18_21_22/0?type=S"
        },
        "time_taken_ms": 49443,
        "expected_duration_ms": 55331,
        "est_wait_to_start_ms": 0,
        "estimated_cost": 0.0009999999351851853,
        "generate_task": false,
        "generated_by": "",
        "artifacts": null,
        "display_only": false,
        "mainline": false
      }
    ]
  ) as object;
}