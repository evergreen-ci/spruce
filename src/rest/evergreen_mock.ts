import * as request from "request";
import * as MockBuild from "./mock_build"
import * as MockData from "./mock_data";
import * as MockLog from "./mock_log";
import * as MockTasks from "./mock_tasks";
import * as MockTests from "./mock_tests";

export class client {
  public username: string;
  public key: string;
  public apiURL: string;
  public uiURL: string;

  constructor(apiURL: string, uiURL: string, username?: string, key?: string) {
    this.username = username;
    this.key = key;
    this.apiURL = apiURL;
    this.uiURL = uiURL;
  }

  public getDistros(callback: request.RequestCallback) {
    callback(null, this.dummySuccessResp(), {});
  }

  public getRecentTasks(callback: request.RequestCallback, verbose?: boolean, lookbackMins?: number, status?: string) {
    callback(null, this.dummySuccessResp(), {});
  }

  public getToken(callback: request.RequestCallback, username?: string, password?: string) {
    callback(null, this.dummySuccessResp(), {});
  }

  public getPatches(callback: request.RequestCallback, username?: string) {
    callback(null, this.dummyPatchesResp(), {});
  }

  public getLogs(callback: request.RequestCallback, id?: string, type?: string, executionNumber?: number) {
    callback(null, this.dummyLogResp(), {});
  }

  public getBuild(callback: request.RequestCallback, id: string) {
    callback(null, this.dummyBuildResp(), {});
  }

  public getTasks(callback: request.RequestCallback, id: string) {
    callback(null, this.dummyTasksResp(), {});
  }

  public getTests(callback: request.RequestCallback, id: string) {
    callback(null, this.dummyTestsResp(), {});
  }

  public getAdminConfig(callback: request.RequestCallback) {
    callback(null, this.dummySuccessResp(), {});
  }

  public setAdminConfig(callback: request.RequestCallback, settings: any) {
    callback(null, this.dummySuccessResp(), {});
  }

  public getBanner(callback: request.RequestCallback) {
    callback(null, this.dummySuccessResp(), {});
  }

  private dummySuccessResp(): request.Response {
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: {},
    };

    return mockResp as request.Response;
  }

  private dummyPatchesResp(): request.Response {
    const bodyAsString = JSON.stringify(MockData.getMockData());
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: bodyAsString,
    };
    return mockResp as request.Response;
  }

  private dummyLogResp(): request.Response {
    const bodyAsString = MockLog.getMockLog();
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: bodyAsString,
    };
    return mockResp as request.Response;
  }

  private dummyBuildResp(): request.Response {
    const bodyAsString = MockBuild.getMockBuild();
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: bodyAsString,
    };
    return mockResp as request.Response;
  }

  private dummyTasksResp(): request.Response {
    const bodyAsString = MockTasks.getMockTasks();
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: bodyAsString,
    };
    return mockResp as request.Response;
  }

  private dummyTestsResp(): request.Response {
    const bodyAsString = MockTests.getMockTests();
    const mockResp: mockResponse = {
      statusCode: 200,
      statusMessage: "",
      body: bodyAsString,
    };
    return mockResp as request.Response;
  }
}

class mockResponse {
  public statusCode: number;
  public statusMessage: string;
  public body: any;
}