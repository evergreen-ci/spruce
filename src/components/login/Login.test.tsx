import * as enzyme from "enzyme";
import * as React from "react";
import * as rest from "../../rest/interface";
import { Login } from "./Login";

describe("Login", () => {
    const wrapper = enzyme.mount(<Login client={rest.EvergreenClient("", "", "", true)} />);

    it("matches snapshot", () => {
      // note: using enzyme.mount here (instead of enzyme.shallow) causes the test suite to hang, so this test needs its own shallow wrapper
      const shallowWrapper = enzyme.shallow(<Login client={rest.EvergreenClient("", "", "", true)} />);
      expect(shallowWrapper).toMatchSnapshot();
    })

    it("clicking login button opens login modal", () => {
      // check that the modal exists, and that it is not currently open
      expect(wrapper.exists("#login-modal")).toBe(true);
      expect(wrapper.state("open")).toBe(false);
      // find the login button and click it
      const loginButton = wrapper.find("#login-button").hostNodes();
      loginButton.simulate("click");
      // check that the modal is now open
      expect(wrapper.state("open")).toBe(true);
    });
    
    it("clicking cancel button closes login modal", () => {
      // find the login button and click it
      const loginButton = wrapper.find("#login-button").hostNodes();
      loginButton.simulate("click");
      // check that the state was set correctly
      expect(wrapper.state("open")).toBe(true);
      // find the cancel button and click it
      const cancelButton = wrapper.find("#cancel-button").hostNodes();
      cancelButton.simulate("click");
      // check that the state was updated correctly
      expect(wrapper.state("open")).toBe(false);
    });
    
    it("clicking submit button closes login modal and saves token locally", () => {
      // find the login button and click it
      const loginButton = wrapper.find("#login-button").hostNodes();
      loginButton.simulate("click");
      // check that the state was set correctly
      expect(wrapper.state("open")).toBe(true);
      // fill in the username and password fields
      wrapper.setState({
        username: "domino.weir", 
        password: "password",
      });
      // find the submit button and click it
      const submitButton = wrapper.find("#submit-button").hostNodes();
      submitButton.simulate("click");
      // check that the state reflects the submit button click
      expect(wrapper.state("submitted")).toBe(true);
      // check that the modal is now closed
      expect(wrapper.state("open")).toBe(false);
      // TODO: check that cookie was included in the response
      // should be mci-token, located in resp.request.response.rawHeaders
    });
})