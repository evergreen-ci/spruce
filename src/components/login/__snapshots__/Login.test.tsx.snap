// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `ShallowWrapper {}`;
